app/Http/Controllers/PortfolioController.php

```diff
@@ -308,11 +452,117 @@ public function storeCurated(Request $request)
+
+        $metadata = $portfolio->metadata;
+
+        abort_if(!$metadata || !isset($metadata['rss_enabled']), 404);
+        abort_unless($metadata['rss_enabled'], 404);
+
+        $account = AccountService::get($user->profile_id);
+        $portfolioUrl = $portfolio->url();
+        $portfolioLayout = $portfolio->profile_layout;
+
+        if(!isset($metadata['posts']) || !count($metadata['posts'])) {
+        	$feed = [];
+        } else {
+        	$feed = Cache::remember(
+        		self::RSS_FEED_KEY . $user->profile_id,
+        		43200,
+        		function() use($portfolio, $portfolioUrl, $portfolioLayout) {
+					return collect($portfolio->metadata['posts'])->map(function($post) {
+						return StatusService::get($post);
+					})
+					->filter()
+					->values()
+					->map(function($post, $idx) use($portfolioLayout, $portfolioUrl) {
+						$ts = now()->parse($post['created_at']);
+						$url = $portfolioLayout == 'album' ? $portfolioUrl . '?slide=' . ($idx + 1) : $portfolioUrl . '/' . $post['id'];
+						return [
+							'title' => 'Post by ' . $post['account']['username'] . ' on ' . $ts->format('D, d M Y'),
+							'description' => $post['content_text'],
+							'pubDate' => date('D, d M Y H:i:s ', strtotime($post['created_at'])) . 'GMT',
+							'url' => $url
+						];
+					})
+					->reverse()
+					->take(10)
+					->toArray();
+        		}
+        	);
+        }
+
+		$now = date('D, d M Y H:i:s ') . 'GMT';
+
+		return response()
+			->view('portfolio.rss_feed', compact('account', 'now', 'feed', 'portfolioUrl'), 200)
+			->header('Content-Type', 'text/xml');
+        return response($feed)->withHeaders(['Content-Type' => 'text/xml']);
+    }
+
+
+    public function getApFeed(Request $request, $username)
+    {
+    	$user = User::whereUsername($username)->first();
+
+        if(!$user) {
+            return view('portfolio.404');
+        }
+
+        $portfolio = Portfolio::whereUserId($user->id)->where('active', 1)->firstOrFail();
+        $metadata = $portfolio->metadata;
+        $baseUrl = config('app.url');
+        $page = $request->input('page');
+
+        $res = [
+        	'@context' => 'https://www.w3.org/ns/activitystreams',
+        	'id' => $portfolio->permalink('.json'),
+        	'type' => 'OrderedCollection',
+        	'totalItems' => isset($metadata['posts']) ? count($metadata['posts']) : 0,
+        ];
+
+        if($request->has('page')) {
+        	$start = $page == 1 ? 0 : ($page * 10 - 10);
+        	$res['id'] = $portfolio->permalink('.json?page=' . $page);
+        	$res['type'] = 'OrderedCollectionPage';
+        	$res['next'] = $portfolio->permalink('.json?page=' . $page + 1);
+        	$res['partOf'] = $portfolio->permalink('.json');
+        	$res['orderedItems'] = collect($metadata['posts'])->slice($start)->take(10)->map(function($p) {
+        		return StatusService::get($p);
+        	})
+        	->filter()
+        	->map(function($p) {
+        		return $p['url'];
```

Comment:
if i'm reading this right, this might break if `url` is not present. maybe fallback to `id` in that case?

Reply:
actually wait, isn't this supposed to be rendering the `orderedItems`? in that case, it should use the `id`